identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp

```diff
@@ -404,8 +404,9 @@
                                             <label class="control-label"><%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "First.name")%>*</label>
                                             <input id="firstNameUserInput" type="text" name="http://wso2.org/claims/givenname" class="form-control"
                                                 <% if (firstNamePII.getRequired() || !piisConfigured) {%> required <%}%>
-                                                <% if (skipSignUpEnableCheck && StringUtils.isNotEmpty(firstNameValue)) { %>
-                                                value="<%= Encode.forHtmlAttribute(firstNameValue)%>" disabled <% } %>
+                                                <% if (skipSignUpEnableCheck && StringUtils.isNotEmpty(firstNameValue)) { %> disabled <% } %>
+                                                <% if (StringUtils.isNotEmpty(firstNameValue)) { %>
```

Comment:
Shouldn't we check whether firstname is required as well here? otherwise, when the first name is not mandatory, an empty value will not be set to the first name input. WDYT?

Reply:
Currently, unless JIT provisioning is enabled (checked with the `skipSignUpEnableCheck` boolean), the value is of the first name is not being set at all, both when it's mandatory and optional.

With this fix, the value will be set if the user has entered a value, and if the user has not entered a value, the existing behaviour, i.e., not setting the value, will be used.